### ISSUE TITLE
BufferSource support

### ISSUE THREAD

**Issue by et1975** [fable-compiler/Fable#1858]
### Description

One of the TS libs I'm trying to use exposes [BufferSource](https://developer.mozilla.org/en-US/docs/Web/API/BufferSource) at the API level, but Fable core doesn't seem to have it defined.

### Repro code

N/A

### Expected and actual results

I think it should be available the Fable core.

**Comment by alfonsogarciacaro** [comment]
Yes, when I tried to unify F# numeric arrays and JS typed arrays for simplification in Fable.Core 3, unfortunately some properties were lost in the way. I recently added some [extensions to array](https://github.com/fable-compiler/Fable/blob/90f098208a95b322592ba80f3d816c306cfd71ae/src/Fable.Core/Fable.Core.Extensions.fs#L11-L20) to include some of these properties, but I'm not sure atm if they're already in the published Fable.Core package.

I need to check which properties are we missing and add them to Fable.Core. In the meantime, it's easy to add such an extension in your own project.
